Makefile

```diff
@@ -65,6 +65,10 @@ DEPS = $(LLVM_EXT_OBJ) $(LIB_CRYSTAL_TARGET)
 CFLAGS += -fPIC $(if $(debug),-g -O0)
 CXXFLAGS += $(if $(debug),-g -O0)
 
+ifeq ($(shell uname -s),Linux)
+  EXPORT_CC ?= $(if $(shell command -v ld.lld),CC="cc -fuse-ld=lld")
+endif
+
```

Comment:
How about just

```makefile
ifeq ($(shell command -v ld.lld >/dev/null && uname -s),Linux)
    CC ?= cc -fuse-ld=lld
endif
```

then leaving off all the stuff down the page.

Reply:
I updated the condition but the CC and other [vars are not exported automatically](https://www.linuxquestions.org/questions/programming-9/exporting-makefile-variables-to-%24-shell-environment-807422/)

That is also a reason why an `EXPORT` variable is used somewhere else in the Makefile.

Reply:
Sounds reasonable. But when overriding `CC`, you'd need to remember to add `-fuse-ld=lld` explicitly...

Reply:
Writing `export CC` works, and is much cleaner.